lib/src/components/chips/chip.dart

```diff
@@ -51,127 +41,171 @@ class ZetaChip extends StatefulWidget {
-          if (states.contains(WidgetState.focused)) {
-            return BorderSide(width: ZetaSpacingBase.x0_5, color: colors.blue.shade50);
-          }
-          return BorderSide(color: colors.borderDefault);
-        }),
-        padding: WidgetStateProperty.all(
-          EdgeInsets.fromLTRB(
-            widget.leading != null ? ZetaSpacingBase.x2_5 : ZetaSpacing.medium,
-            ZetaSpacing.none,
-            widget.trailing != null ? ZetaSpacingBase.x2_5 : ZetaSpacing.medium,
-            ZetaSpacing.none,
-          ),
-        ),
-      ),
-      child: Row(
-        mainAxisSize: MainAxisSize.min,
-        children: [
-          if (leading != null) _renderLeading(foregroundColor),
-          Text(widget.label),
-          if (widget.trailing != null)
-            IconTheme(
-              data: IconThemeData(
-                color: foregroundColor,
-                size: ZetaSpacing.xl_1,
+
+    return SelectionContainer.disabled(
+      child: widget.draggable
+          ? Draggable(
+              feedback: Material(
+                color: Colors.transparent,
+                child: child(colors, foregroundColor, isDragging: true),
               ),
-              child: widget.trailing!,
+              childWhenDragging: const SizedBox(),
+              data: widget.data,
+              onDragCompleted: widget.onDragCompleted,
+              child: child(colors, foregroundColor),
+            )
+          : child(colors, foregroundColor),
+    );
+  }
+
+  ValueListenableBuilder<Set<WidgetState>> child(ZetaColors colors, Color foregroundColor, {bool isDragging = false}) {
+    return ValueListenableBuilder(
+      valueListenable: _controller,
+      builder: (context, states, child) {
+        final double iconSize = selected ? ZetaSpacing.xl_2 : ZetaSpacing.none;
+
+        return InkWell(
+          statesController: _controller,
+          borderRadius: widget.rounded ? ZetaRadius.full : ZetaRadius.none,
+          onTap: () {
+            if (widget.selected != null) {
+              setState(() => selected = !selected);
+              widget.onToggle?.call(selected);
+            } else {
+              widget.onTap?.call();
+            }
+          },
+          child: AnimatedContainer(
+            duration: Durations.short3,
+            height: ZetaSpacing.xl_5,
+            padding: EdgeInsets.fromLTRB(
+              widget.leading != null ? ZetaSpacingBase.x2_5 : ZetaSpacing.medium,
+              0,
+              widget.trailing != null ? ZetaSpacingBase.x2_5 : ZetaSpacing.medium,
+              0,
             ),
-        ].divide(const SizedBox.square(dimension: ZetaSpacing.small)).toList(),
-      ),
+            decoration: BoxDecoration(
+              color: () {
+                if (states.contains(WidgetState.disabled)) {
+                  return colors.surfaceDisabled;
+                }
+                if (selected) {
+                  return colors.cool.shade90;
```

Comment:
Selected shade feels a bit off. I think it should be colors.surfaceDefaultInverse